core-bundle/src/EventListener/MergeHttpHeadersListener.php

```diff
@@ -101,11 +102,24 @@ public function reset(): void
     /**
      * Fetches and stores HTTP headers from PHP.
      */
-    private function fetchHttpHeaders(): void
+    private function fetchHttpHeaders(Request $request): void
     {
         $headers = $this->headerStorage->all();
+        $session = $request->hasSession() ? $request->getSession() : null;
+        $deprectatedHeaders = array_filter(
+            $headers,
+            static function ($header) use ($session): bool {
+                // Ignore Set-Cookie header set by PHP when using NativeSessionStorage
+                if ($session && str_starts_with($header, "Set-Cookie: {$session->getName()}=")) {
+                    return false;
+                }
+
+                // Ignore X-Powered-By header
+                return !str_starts_with($header, 'X-Powered-By:');
+            },
+        );
 
-        if ([] !== $headers) {
+        if ([] !== $deprectatedHeaders) {
             trigger_deprecation('contao/core-bundle', '5.3', 'Using the PHP header() function to set HTTP headers has been deprecated and will no longer work in Contao 6. Use the response object instead.');
```

Comment:
```suggestion
            trigger_deprecation('contao/core-bundle', '5.3', 'Using the PHP header() function to set HTTP headers has been deprecated and will no longer work in Contao 6. Use the response object instead. Headers used: %s', implode(', ', $deprectatedHeaders));
```
Should we include the headers in the message, so that it’s easier to find where the deprecation comes from?